src/main/kotlin/com/autonomousapps/internal/utils/gradleStrings.kt

```diff
@@ -141,50 +141,46 @@ internal fun ComponentIdentifier.resolvedVersion(): String? = when (this) {
  * Given [Configuration.getDependencies], return this dependency set as a set of identifiers, per
  * [ComponentIdentifier.toIdentifier].
  */
-internal fun DependencySet.toIdentifiers(
-  metadataSink: MutableMap<String, Boolean> = mutableMapOf()
```

Comment:
I think the idea with the metadata can not work like it was here. You would need the metadata for each dependency. E.g. if you have both:

```
api("org.apache.commons:commons-collections4:4.4")
testFixturesApi("org.apache.commons:commons-collections4:4.4")
```

The two declaration would have different metadata. But the approach here only stores one "Metadata" for `org.apache.commons:commons-collections4:4.4`.

It also feels to me like Gradle's dependency resolution should be used here to give you the result (i.e. to which Jar on the other side will the dependency resolve).
If you start storing attributes/capabilities here yourself and pass them around you will somehow end up re-implementing Gradle's variant matching to cover all the cases.

Reply:
I think, if you want to get to a working solution here later, you should investigate how to do this:

> It also feels to me like Gradle's dependency resolution should be used here to give you the result (i.e. to which Jar on the other side will the dependency resolve).

It's not the goal of this PR to solve this. But feel free to ping me if you get to it later. Happy to discuss/explore this.

Reply:
I think one of the problems was doing this in the context of _declarations_, right? At that point, I don't want to _resolve_ any dependencies at all. I just want to know what the user _declared_.